src/extensions/tos.ts

```diff
@@ -0,0 +1,120 @@
+
+export async function acceptAppDeveloperTOS(
+  projectId: string,
+  tosVersion: string,
+  instanceId: string = ""
+): Promise<AppDevTOS> {
+  const res = await apiClient.post<
+    { name: string; instanceId: string; version: string },
+    AppDevTOS
+  >(`/projects/${projectId}/appdevtos:accept`, {
+    name: `project/${projectId}/appdevtos`,
+    instanceId,
+    version: tosVersion,
+  });
+  return res.body;
+}
+
+export async function getPublisherTOSStatus(projectId: string): Promise<PublisherTOS> {
+  const res = await apiClient.get<PublisherTOS>(`/projects/${projectId}/publishertos`);
+  return res.body;
+}
+
+export async function acceptPublisherTOS(
+  projectId: string,
+  tosVersion: string
+): Promise<PublisherTOS> {
+  const res = await apiClient.post<{ name: string; version: string }, PublisherTOS>(
+    `/projects/${projectId}/publishertos:accept`,
+    {
+      name: `project/${projectId}/publishertos`,
+      version: tosVersion,
+    }
+  );
+  return res.body;
+}
+
+export async function acceptLatestPublisherTOS(
+  options: { force?: boolean; nonInteractive?: boolean },
+  projectId: string
+): Promise<PublisherTOS> {
+  logger.debug(`Checking if latest publisher TOS has been accepted by ${projectId}...`);
+  const currentAcceptance = await getPublisherTOSStatus(projectId);
+  if (currentAcceptance.lastAcceptedVersion === currentAcceptance.latestTosVersion) {
+    logger.debug(
+      `Latest version of publisher TOS is ${currentAcceptance.lastAcceptedVersion}, already accepted.`
+    );
+    return currentAcceptance;
+  } else {
+    // Display link to TOS, prompt for acceptance
+    const tosLink = extensionsTosUrl("publisher");
+    logger.info(
+      `To continue, you must accept the [Extensions publisher terms of service](${tosLink})`
+    );
+    if (
+      await confirm({
+        ...options,
+        message: "Do you accept the Extensions publisher terms of service?",
+      })
+    ) {
+      return acceptPublisherTOS(projectId, currentAcceptance.latestTosVersion);
+    }
+    throw new FirebaseError("Command terminated becuase publisher TOS was not accepted");
+  }
+}
+
+export async function acceptLatestAppDeveloperTOS(
+  options: { force?: boolean; nonInteractive?: boolean },
+  projectId: string,
+  instanceId: string = ""
+): Promise<AppDevTOS> {
+  logger.debug(
+    `Checking if latest AppDeveloper TOS has been accepted by ${projectId}/${instanceId}...`
+  );
+  const currentAcceptance = await getAppDeveloperTOSStatus(projectId);
+  if (currentAcceptance.lastAcceptedVersion === currentAcceptance.latestTosVersion) {
+    logger.debug(
+      `Latest version of app developer TOS is ${currentAcceptance.lastAcceptedVersion}, already accepted.`
+    );
+    return currentAcceptance;
+  } else {
+    // Display link to TOS, prompt for acceptance
+    const tosLink = extensionsTosUrl("appdev");
+    logger.info(
+      `To continue, you must accept the [Extensions app developer terms of service](${tosLink})`
```

Comment:
Possibly nit: Capitalize and use full name "Firebase Extensions App Developer Terms of Service" and "Firebase Extensions Publisher Terms of Service" I think.

Reply:
Was assuming that this is another case of 'Not Everything Needs To Be In Caps Case'

Reply:
I think this is one case that it's important To Be In Caps since it refers to the legal document with the same title, not some abstract terms.